cfgov/v1/util/queries.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import ref
+
+from ..models.learn_page import AbstractFilterPage
+
+logger = logging.getLogger(__name__)
+
+
+class FilterableQuery(object):
+
+    def __init__(self, form, hostname, page=None):
+        self.form = form
+        self.hostname = hostname
+        self.page = page
+
+    def update_categories(self):
+        """ This is a (hopefully) temporary solution for dealing w/ the fact
+        that we show Blog and Reports as options for filtering, but
+        aren't categories themselves. Rather, they consist of sub-categories,
+        but since we aren't showing those sub-categories to the end user,
+        selecting the parent category is equivalent to all of them
+        getting checked. The mapping of Blog and Reports to their
+        respective categories exists in cfgov/v1/util/ref.py
+        """
+        categories = self.form.cleaned_data.get('categories', [])
+        if 'blog' in categories:
+            self.form.cleaned_data['categories'].remove('blog')
+            for x in ref.categories[1][1]:
```

Comment:
This seems prone to breaking, especially the hardcoded `[8][1]`. I see why it's tricky, because the form gives you back something like `research-reports` and there's no non-hacky way to look up 'Research Report' in the reference without hardcoding it. Is there a reason that we can't change [this line](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/util/ref.py#L146) to `"Research Report"` so that it's at least possible to programmatically look up and map it to its subcategories [here](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/util/ref.py#L97)?

From a design point of view, I wonder if it might be better to modify modify `FilterableListForm.generate_query` to do this subcategory expansion. There'd need to be some extra logic [here](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/forms.py#L185) that expanded the category. This makes sense to me as the method named `generate_query` would imply "this method generates a Q object that can be used to look up pages as specified by this form's values". If we're implementing custom logic to expand subcategories (because otherwise the query the form generates actually doesn't work to get, e.g. the blog pages), then should that be part of the form class?